yellowbrick/classifier/class_balance.py

```diff
@@ -168,3 +175,171 @@ def class_balance(model, X, y=None, ax=None, classes=None, **kwargs):
 
     # Return the axes object on the visualizer
     return visualizer.ax
+
+
+##########################################################################
+## Class Prediction Error Chart
+##########################################################################
+
+class ClassPredictionError(ClassificationScoreVisualizer):
+    """
+    Class Prediction Error chart that shows the support for each class in the
+    fitted classification model displayed as a stacked bar.  Each bar is
+    segmented to show the distribution of predicted classes for each
+    class. It is initialized with a fitted model and generates a
+    class prediction error chart on draw.
+
+    Parameters
+    ----------
+    ax: axes
+        the axis to plot the figure on.
+    model: estimator
+        Scikit-Learn estimator object. Should be an instance of a classifier,
+        else ``__init__()`` will raise an exception.
+    classes: list
+        A list of class names for the legend. If classes is None and a y value
+        is passed to fit then the classes are selected from the target vector.
+    kwargs: dict
+        Keyword arguments passed to the super class. Here, used
+        to colorize the bars in the histogram.
+    Notes
+    -----
+    These parameters can be influenced later on in the visualization
+    process, but can and should be set as early as possible.
+    """
+
+    def score(self, X, y=None, **kwargs):
+        """
+        Generates a 2D array where each row is the count of the
+        predicted classes and each column is the true class
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        Returns
+        -------
+
+        ax : the axis with the plotted figure
+        """
+
+        # We're replying on predict to raise NotFitted
+        y_pred = self.predict(X)
+
+        y_type, y_true, y_pred = _check_targets(y, y_pred)
+
+        if y_type not in ("binary", "multiclass"):
+            raise YellowbrickValueError("%s is not supported" % y_type)
+
+        indices = unique_labels(y_true, y_pred)
+
+        if len(self.classes_) != len(indices):
+            raise ModelError(
+                'Please correct label mismatch.')
```

Comment:
So I'm trying to reason about this outloud: first, we know that `self.classes_` is either specified by the user as a list of labels or it is extracted from [`self.estimator.classes_`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/base.py#L108). We should note that we do no further checks on that, so if `self.classes_` is None, then we're in trouble, but I don't think that's your problem, I think that's a problem for `ClassificationScoreVisualizer`. 

If `self.classes_` is supplied by the user:

- if `len(self.classes_) < len(indices)` it could be that the user is trying to filter labels, but this is currently not implemented. 
- if `len(self.classes_) > len(indices)` then potentially the user-supplied bad class information or potentially the reason below for the same condition.

If `self.classes_` is derived from the underlying estimator:

- if `len(self.classes_) < len(indices)` it could be that the estimator does not know about classes in y_true, but presumably this would have errored during the computation of y_pred and will get caught in `_check_targets` 

- if `len(self.classes_) > len(indices)` it could be that `y` and `y_pred` are missing a class that was contained in `y_train`. In that case, we don't know which class is missing without knowing if we have label encoded data or not. The bad thing that could happen is that we mislabel the classes, which would be ultra bad.  We have two choices here, raise an exception (current choice) or make a choice and warn the user. Making a choice would likely involve comparing the contents of `self.classes_` with `indices`. 

Does that reasoning make sense? 

Currently I propose the following:

```python
if len(self.classes_) > len(indices):
    raise ModelError('y and y_pred contain zero values of one of the specified classes') 
elif len(self.classes_) < len(indices):
    raise NotImplementededError('filtering classes is currently not supported') 
```

Thoughts?

Reply:
Your logic seems fine...  While fooling around today, I accidentally raised this error.  

I had built a classification dataset (sklearn - make_classification) with 5 classes ['a','b','c','d','e'] however when I ran ClassPredictionError I only specified 4 classes

```python 
viz = ClassPredictionError(RandomForestClassifier(), classes=['a','b','c','d']
viz.fit(X_train, y_train)
viz.score(X_test, y_test)
viz.poof()

---------------------------------------------------------------------------
ModelError                                Traceback (most recent call last)
<ipython-input-8-dc1390c3a6b8> in <module>()
      4                                   classes=['a', 'b', 'c', 'd'])
      5 visualizer.fit(X_train, y_train)
----> 6 visualizer.score(X_test, y_test)
      7 visualizer.poof(outpath='heatmap.png')
      8 plt.show()

~/portfolio/yellowbrick/yellowbrick/classifier/class_balance.py in score(self, X, y, **kwargs)
    239         if len(self.classes_) != len(indices):
    240             raise ModelError(
--> 241                 'Please correct label mismatch.')
    242 
    243         # Create a table of scores whose rows are the true classes

ModelError: Please correct label mismatch.


```